spring-cloud-sleuth-core/src/test/java/org/springframework/cloud/sleuth/instrument/hystrix/JavanicaITest.java

```diff
@@ -0,0 +1,125 @@
+package org.springframework.cloud.sleuth.instrument.hystrix;
+
+import static org.assertj.core.api.BDDAssertions.then;
+
+import java.util.concurrent.atomic.AtomicReference;
+
+import org.junit.After;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.SpringApplicationConfiguration;
+import org.springframework.cloud.netflix.hystrix.EnableHystrix;
+import org.springframework.cloud.sleuth.Span;
+import org.springframework.cloud.sleuth.TraceManager;
+import org.springframework.cloud.sleuth.instrument.DefaultTestAutoConfiguration;
+import org.springframework.cloud.sleuth.trace.TraceContextHolder;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.context.annotation.EnableAspectJAutoProxy;
+import org.springframework.test.context.junit4.SpringJUnit4ClassRunner;
+
+import com.jayway.awaitility.Awaitility;
+import com.netflix.hystrix.contrib.javanica.annotation.HystrixCommand;
+
+@RunWith(SpringJUnit4ClassRunner.class)
+@SpringApplicationConfiguration(classes = {
+		JavanicaITest.JavanicaITestConfiguration.class })
+public class JavanicaITest {
+
+	@Autowired JavanicaClass javanicaClass;
+	@Autowired JavanicaDelegation javanicaDelegation;
+	@Autowired TraceManager traceManager;
+
+	@Test
+	public void should_set_span_on_an_hystrix_command_annotated_method() {
+		final Span span = givenASpanInCurrentThread();
```

Comment:
mentioned offline, and understand this is a POV vs other POV thing, but let's consider this class has about 100 lines of stuff, but only one test. The test is indirected by splitting into BDD methods. This is an optimization of BDD style over actual usage. In my experience, tests that represent usage are great ways to teach people how to use the code and also where to plop edge cases into. This particular method would be imho less complex and more compelling inlined.

Intentionally took off the "this."s because they are horrendous to read imho.

```java
Span span = traceManager.startSpan("existing").getSpan();

// it is important to note why we need to invoke this..
traceManager.continueSpan(span);

// invoke a hystrix call whose only task is to catch its span
spanCatcher.invoke();

// make sure this is a child span?
assertThat(spanCatcher.takeSpan())
   .hasTraceId(span.getId())
   .hasSpanName("what we actually expect");
```